core/server/src/test/java/alluxio/master/journal/ProtoBufJournalFormatterTest.java

```diff
@@ -19,4 +29,18 @@
   protected JournalFormatter getFormatter() {
     return new ProtoBufJournalFormatter();
   }
+
+  @Test
+  public void truncatedEntryDoesntFail() throws Exception {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    ProtoBufJournalFormatter formatter = new ProtoBufJournalFormatter();
+    formatter.serialize(JournalEntry.newBuilder().setCompleteFile(
```

Comment:
No, the restart process goes

1. Update master state to reflect checkpoint + all additional journal entries
2. Write a new checkpoint file representing the current state
3. Overwrite the old checkpoint with the new checkpoint.